metrics/engine/ingester.go

```diff
@@ -87,6 +91,31 @@ func (oi *outputIngester) flushMetrics() {
 				oi.metricsEngine.markObserved(sm.Metric)
 				sm.Metric.Sink.Add(sample)
 			}
+
+			if _, ok := oi.seenTimeSeries[sample.TimeSeries]; !ok {
+				oi.seenTimeSeries[sample.TimeSeries] = struct{}{}
+			}
 		}
 	}
+
+	oi.warnTimeSeriesLimit()
+}
+
+func (oi *outputIngester) warnTimeSeriesLimit() {
+	if len(oi.seenTimeSeries) <= oi.timeSeriesLimit {
+		return
+	}
+
+	// TODO: suggest to use as an alternative option the Metadata API
+	// when it will be available (e.g. move high-cardinality tags as Metadata)
+	oi.logger.Warnf("The test has generated %d time series, it is higher than suggested limit value, "+
+		"it could generate a high memory usage. "+
+		"You may consider to not use high-cardinality values like IDs as tags "+
+		"or if you need them in the URL then use the URL-grouping feature. "+
+		"Check out https://k6.io/docs/using-k6/tags-and-groups documentation.", len(oi.seenTimeSeries))
```

Comment:
```suggestion
	oi.logger.Warnf("The test has generated %d time series, which is higher than the suggested limit (%d), "+
		"and could cause high memory usage. "+
		"Consider not using high-cardinality values like IDs as tags "+
		"or if you need them in the URL then use the URL-grouping feature. "+
		"See https://k6.io/docs/using-k6/tags-and-groups for details.", oi.timeSeriesLimit, len(oi.seenTimeSeries))
```

It's not properly formatted, so don't just accept the suggestion as is.